src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1524,6 +1524,12 @@ public sealed override bool IsTupleCompatible(out int tupleCardinality)
             return false;
         }
 
+        // PROTOTYPE: Should be abstract.
+        internal virtual NamedTypeSymbol AsNativeInt() => throw ExceptionUtilities.Unreachable;
+
```

Comment:
Not sure why we have/want this.

Reply:
`AsNativeInt()` is used to generate the `NamedTypeSymbol` instance for `nint` (or `nuint`) from a `NamedTypeSymbol` instance for `System.IntPtr` (or `System.UIntPtr`).

See examples of `compilation.GetSpecialType(SpecialType.System_IntPtr).AsNativeInt()`.

---
In reply to: [334622344](https://github.com/dotnet/roslyn/pull/39157#discussion_r334622344) [](ancestors = 334622344)